tests/v2/integration/rtbs/rtbs_test.go

```diff
@@ -120,7 +118,14 @@ func (p *RTBTestSuite) TestPRTBRoleTemplateInheritance() {
 		})
 	require.NoError(p.T(), err)
 
-	err = users.AddProjectMember(client, p.project, p.testUser, rtA.ID)
+	err = users.AddProjectMember(client, p.project, p.testUser, rtA.ID, []*authzv1.ResourceAttributes{
+		{
+			Verb:      "get",
+			Resource:  "secrets",
+			Name:      secret.Name,
+			Namespace: createdNamespace.Name,
+		},
```

Comment:
This is a little repetetive, since most of this rule is already in the role template. Could `AddProjectMember` just infer this rule from the role template?

Reply:
I didn't want to recursively gather the role templates from a "root" role template in a different design where we'd use it to determine what CRs/CRBs would be created where. That seemed too complex.

But doing the same thing to collect a bunch of rules (ResourceAttributes) to test with the Self SARs might be do-able.. Not sure if we'd want to test all combination of verbs/namespace/names, etc or just one for each RoleTemplate (and inherited role template). eg: If we had something like:

```yaml
apiVersion: management.cattle.io/v3
kind: RoleTemplate
...
rules:
- apiGroups:
  - ""
  resourceNames:
  - secret-1
  - secret-2
  resources:
  - secrets
  verbs:
  - get
  - delete
```

then we could do a Self SAR check with only 

```go
{
	Verb:      "get",
	Resource:  "secrets",
	Name:      "secret-1",
	Namespace: "???"
},
```

For namespaced resources we'd need to find a namespace that belongs in the given project. Yeah let me think about it.

Reply:
Sounds like it is simplest to go with what you already have. Thanks for looking into it.